### ISSUE TITLE
Sig value not showing

### ISSUE THREAD

**Issue by bryan824** [silkenweb/silkenweb#40]
I might be doing something wrong here. But to reproduce this, we can take [counter-list](https://github.com/silkenweb/silkenweb/blob/aadbca585a9bc39d6c767b05a315546372c85065/examples/counter-list/src/main.rs#L17-L33) as an example.

If we change the main function to the following:

```rust
fn main() {
    let list = Rc::new(MutableVec::new());
    list.lock_mut().push_cloned(Counter);
    mount(
        "app",
        div()
            .text("How many counters would you like?")
            // .child(
            //     div()
            //         .child(pop_button(list.clone()))
            //         .text(Sig(list.signal_vec().len().map(|len| format!("{len}"))))
            //         .child(push_button(list.clone())),
            // )
            .child(hr())
            .child(div().children_signal(list.signal_vec().map(|_| define_counter()))),
    );
}
```

I would assume there should be a counter that is defined and initialized with 0. but it ends up only showing two buttons and the counter value between is missing.

Thank you!

**Comment by simon-bourne** [comment]
Hi, thanks for the report! I need to investigate further, but here's what I know so far:

This is happening because `list` is dropped after `main` finishes, which for some reason stops the first value of the signal getting through. The click handlers in `push_button` and `pop_button` hold a reference to the list which is why this works when they are present.

I don't have time to investigate today, but will take a look tomorrow.

**Comment by simon-bourne** [comment]
This is fixed by #41 

Assuming the "counter-list" was a cut down example, does this fix your original problem?

**Comment by bryan824** [comment]
Yes, the update fixes my original problem. Here is the original code and I could not see the info log before.

```rust
  .children_signal(this.accounts_signal().map(|a| {
      div()
          .child(
              span()
                  .text(a.name.clone()),
          )
          .child(input().value(a.balance.to_string()).on_input(|_el, ev| {
              log::info!("{}", ev.value());
          }))
  }))
```

now it works great. Thank you!
